### PR TITLE
Staging for 0.7.2 release.

### PR DESCRIPTION
**Pull request by pranaysashank**
Also closes #489.

### CODE REVIEW

**Comment by harendra-kumar** [comment]
I see these regressions in the `linear` benchmark:

```
Linear(cpuTime)
Benchmark                                          default(0)(μs) default(1) - default(0)(%)
-------------------------------------------------- -------------- --------------------------
serially/elimination/minimum                                42.06                    +750.99
serially/elimination/maximum                                48.49                    +741.99
serially/mixedX4/filter-drop                                44.06                    +655.04
serially/elimination/maximumBy                              63.08                    +550.93
serially/elimination/minimumBy                              62.81                    +476.79
serially/filteringX4/insertBy                              190.61                    +388.42
serially/pipesX4/zip                                       270.12                     +35.77
serially/mixedX4/take-map                                  126.64                     +22.75
serially/filteringX4/intersperse                          3142.86                     +11.37
serially/concat-serial/concatMapRepl (x/4,4)               623.59                      +8.59
serially/foldable/foldrElem                                 46.98                      +5.20
```

And many improvements as well, including some big ones:

```
serially/mixedX4/drop-scan                                 406.50                    -477.08
serially/filteringX4/dropWhile-false                       191.64                    -503.01
serially/fold-multi-stream/isPrefixOf                      889.08                   -1305.17
serially/fold-multi-stream/stripPrefix                     893.95                   -1314.03
serially/fold-multi-stream/isSubsequenceOf                 944.91                   -1403.38
```

Not sure what caused these.

All other benchmarks do not show any significant difference.

File: Changelog.md [hunk 1]
```diff
@@ -1,8 +1,14 @@
-## Unreleased
+## 0.7.2
+
 ### Bug Fixes
 
+* Fix a bug in the `Applicative` and `Functor` instances of the `Fold`
+  data type.
+* Fix a bug that caused the streamly builds to fail on 32-bit machines.
```
Comment (harendra-kumar) [inline Changelog.md:7]: Looking at this section looks like there are many bug fixes. We can put the build related fixes in a separate "Build Issues" section.

**Comment by pranaysashank** [comment]
> Raise a separate issue for tracking the benchmark regressions reported in this PR.

I think #506 is enough for that?

**Comment by harendra-kumar** [comment]
> I think #506 is enough for that?

This is different than 506 because the cause of the regression is entirely different. In 506 the regressions are because of the compiler change. Here the regression is because of the code changes in streamly, compiler being the same. We should not conflate the two.

**Pull request merged**